gmso/core/topology.py

```diff
@@ -625,6 +625,143 @@ def is_typed(self, updated=False):
+            'dihedrals' : check for status of all topology.dihedrals
+            'impropers' : check for status of all topology.impropers
+        full : bool, optional, default=True
+            If True, check if the checked group s fully typed,
+            else, check if the group is at least partially typed.
+
+        Returns
+        -------
+        bool
+            Status of the check
+        Notes
+        -----
+        `self._type` is set to True as long as the Topology is at least
+        partially typed.
+        """
+        if not updated:
+            self.update_connection_types()
+            self.update_atom_types()
+
+        typed_status = {
+            "sites": lambda top: all(site.atom_type for site in top._sites),
+            "bonds": lambda top: all(bond.bond_type for bond in top._bonds),
+            "angles": lambda top: all(
+                angle.angle_type for angle in top._angles
+            ),
+            "dihedrals": lambda top: all(
+                dihedral.dihedral_type for dihedral in top._dihedrals
+            ),
+            "impropers": lambda top: all(
+                improper.improper_type for improper in top._impropers
+            ),
+        }
+
+        if group == "topology":
+            result = list()
+            for subgroup in typed_status:
+                result.append(typed_status[subgroup](self))
+            return all(result)
+        elif group in typed_status:
+            return typed_status[group](self)
+        else:
+            raise ValueError(
+                f"Could not check typing status of {group}. "
+                "Available options: 'topology', 'sites', 'bonds', "
+                "'angles', 'dihedrals', 'impropers'."
+            )
+
+    def get_untyped(self, group):
+        """Get untyped objects of the Topology
+
+        Parameters
+        ----------
+        group : {'sites', 'bonds', 'angles', 'dihedrals', 'impropers', 'all'}
+            The group of object to be checked. The 'topology' option will return
+            all untyped object of the topology.
+
+        Returns
+        -------
+        untyped : dict
+            Dictionary of all untyped object, key of the dictionary correspond to
+            object groups
+        """
+        untyped = dict()
+        untyped_extractors = {
+            "sites": self._get_untyped_sites,
+            "bonds": self._get_untyped_bonds,
+            "angles": self._get_untyped_angles,
+            "dihedrals": self._get_untyped_dihedrals,
+            "impropers": self._get_untyped_impropers,
+        }
+        if group == "topology":
+            for subgroup in untyped_extractors:
+                untyped.update(untyped_extractors[subgroup])
+        elif group in untyped_extractors:
+            untyped = untyped_extractors[group]()
+        else:
+            raise ValueError(
+                f"Cannot get untyped {group}. "
+                "Available options: 'sites', 'bonds', "
```

Comment:
Instead of hardcoding these groups here, lets use the dict `untyped_extractors` keys here.
```suggestion
                f"Available options: {untyped_extractors.keys()}."
```

Might need to do a list comprehension to better clean it up

```python3
f"available options: {[key for key in untyped_extractors.keys()]}"